File: Creational/Builder/JavaScript/js/BurgerMealBuilder.js
// js/BurgerMealBuilder.js
import { MealBuilder } from './MealBuilder.js';
import { Meal } from './Meal.js';

export class BurgerMealBuilder extends MealBuilder {
    constructor() {
        super();
        this.meal = new Meal();
    }

    buildMainItem() {
        this.meal.setMainItem('Burger');
    }

    buildSideItem() {
        this.meal.setSideItem('Fries');
    }

    buildDrink() {
        this.meal.setDrink('Coke');
    }

    buildToy() {
        this.meal.setToy('Toy Car');
    }

    getMeal() {
        return this.meal;
    }
}

File: Creational/Builder/Java/src/main/java/com/fastfood/BurgerMealBuilder.java
package main.java.com.fastfood;

public class BurgerMealBuilder implements MealBuilder {
    private Meal meal;

    public BurgerMealBuilder() {
        this.meal = new Meal();
    }

    @Override
    public void buildMainItem() {
        meal.setMainItem("Burger");
    }

    @Override
    public void buildSideItem() {
        meal.setSideItem("Fries");
    }

    @Override
    public void buildDrink() {
        meal.setDrink("Coke");
    }

    @Override
    public void buildToy() {
        meal.setToy("Toy Car");
    }

    @Override
    public Meal getMeal() {
        return this.meal;
    }
}
